vcd/resource_vcd_org.go

```diff
@@ -74,6 +74,67 @@ func resourceOrg() *schema.Resource {
 				Default:     true,
 				Description: "True if this organization is allowed to share catalogs.",
 			},
+			"vapp_lease": &schema.Schema{
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
+				MaxItems:    1,
+				Description: "Defines lease parameters for vApps created in this organization",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"maximum_runtime_lease_in_sec": &schema.Schema{
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Description:  "How long vApps can run before they are automatically stopped (in seconds). 0 means never expires",
+							ValidateFunc: validateIntLeaseSeconds(), // Lease can be either 0 or 3600+
+						},
+						"power_off_on_runtime_lease_expiration": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+							Description: "When true, vApps are powered off when the runtime lease expires. " +
+								"When false or missing, vApps are suspended when the runtime lease expires",
+						},
+						"maximum_storage_lease_in_sec": &schema.Schema{
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Description:  "How long stopped vApps are available before being automatically cleaned up (in seconds). 0 means never expires",
+							ValidateFunc: validateIntLeaseSeconds(), // Lease can be either 0 or 3600+
+						},
+						"delete_on_storage_lease_expiration": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
```

Comment:
Is it on purpose that `power_off_on_runtime_lease_expiration` and `delete_on_storage_lease_expiration` are optional and have defaults, but `maximum_runtime_lease_in_sec` and `maximum_storage_lease_in_sec` don't have defaults, but are also optional?

What would be the meaning of such definition (which seem to be allowed):

```
vapp_lease {
power_off_on_runtime_lease_expiration = true
delete_on_storage_lease_expiration = true
}
```

Or such:


```
vapp_lease {
}
```

Reply:
The lease amounts can change depending on which vCD version you use.

If you set any values within `vapp_lease` block (or if you put an empty block), the result for all the lease amounts will be 0 (= never expires).

If you don't set the block at all, the result will be whatever default the vCD version uses.

Reply:
> If you set any values within vapp_lease block (or if you put an empty block), the result for **all the lease amounts will be 0 (= never expires)**.

Do you mean that in my first example where I don't set `maximum_runtime_lease_in_sec` it would be forced to 0? That's not what the user would expect, but also not what the docs are saying:

> maximum_runtime_lease_in_sec - (Optional) - How long vApps can run before they are automatically stopped (in seconds). 0 means never expires. Values accepted from 3600+. - **Default is 604800 (7 days) but may vary depending on vCD version**

Reply:
Made fields in lease sections mandatory. Docs updated